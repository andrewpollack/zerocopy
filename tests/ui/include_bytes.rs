// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#[macro_use]
extern crate zerocopy;

fn main() {
    // Should fail because the file is 4 bytes long, not 8.
    let _: u64 = include_bytes!("../include_bytes.data");
    // Should fail because `UnsafeCell<u32>: !FromBytes`.
    let _: core::cell::UnsafeCell<u32> = include_bytes!("../include_bytes.data");
}
