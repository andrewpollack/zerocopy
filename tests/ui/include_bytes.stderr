error[E0277]: the trait bound `UnsafeCell<u32>: FromBytes` is not satisfied
  --> tests/ui/include_bytes.rs:12:42
   |
12 |     let _: core::cell::UnsafeCell<u32> = include_bytes!("../include_bytes.data");
   |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromBytes` is not implemented for `UnsafeCell<u32>`
   |
   = help: the following other types implement trait `FromBytes`:
             ()
             F32<O>
             F64<O>
             I128<O>
             I16<O>
             I32<O>
             I64<O>
             MaybeUninit<T>
           and 35 others
note: required by a bound in `main::transmute`
  --> tests/ui/include_bytes.rs:12:42
   |
12 |     let _: core::cell::UnsafeCell<u32> = include_bytes!("../include_bytes.data");
   |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `main::transmute`
   = note: this error originates in the macro `transmute` which comes from the expansion of the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)
